server/ast/expr.go

```diff
@@ -532,3 +543,43 @@ func nodeExpr(node tree.Expr) (vitess.Expr, error) {
 		return nil, fmt.Errorf("unknown expression: `%T`", node)
 	}
 }
+
+func typeConvertType(typ tree.ResolvableTypeReference) (*vitess.ConvertType, error) {
```

Comment:
Three things here:

1) I see that this function's body was taken from `server/ast/column_table_def.go`. In that case, we absolutely want this to be its own function, since it's used in more than 1 spot. That also means that we need to remove the code in `nodeColumnTableDef` and have it call this function instead, else we get code duplication, and I _absolutely_ want to avoid that whenever possible. There are times where a node can contain different information depending on its contextual usage, but this is very much not the case here (see `*tree.AllColumnsSelector` for an example). I know that this will slow down development, but we don't have any reason to rush just yet, so I want to spend as much time as possible getting things in as good of a state as possible, so it's not an unmanageable mess later on.
2) This name is inconsistent with the naming scheme. All functions that convert from a Postgres `tree` type to a `vitess` type starts with `node`, and the remaining name mirrors that of the input type. So this should be named `nodeResolvableTypeReference`. The name is chosen for ease of discoverability, as if you know you have some Postgres type and want to parse it, then you know it's just `nodeX`. If `nodeX` doesn't exist, then you know you need to make it. Out of the hundreds of functions, nobody would be able to guess that `typeConvertType` is for `ResolvableTypeReference` (could be for `tree.TypedExpr` just based on the name). I know I probably sound pedantic, but I'm just trying to minimize a class of errors that we've run into time and time again.
3) This should go in its own file, named `resolvable_type_reference.go`. The `expr` file is mainly for `tree.expr`, and the only other functions that are allowed in there are those that function as wrappers over `nodeExpr`. Again, this is for discoverability and to reduce one file growing to an insane size, and it makes it easier to find the functionality for any given node without having to use an IDE's "Go To" functionality.

Reply:
You're a pedant but you're /our/ pedant